test_common/jest.config.js

```diff
@@ -37,27 +15,15 @@ const config = {
       statements: 100,
     },
   },
-  moduleDirectories: ['node_modules', 'client', 'server'],
-  moduleFileExtensions: ['js', 'feature', 'ts'],
-  testEnvironment: 'jsdom',
-  testMatch: ['**/*.spec.ts', '**/*.steps.ts'],
-  testPathIgnorePatterns: ['/e2e/'],
-  transform: {
-    '^.+\\.(ts|tsx)$': 'ts-jest',
-    '^.+\\.(js|jsx|ts|tsx)$': [
-      'babel-jest',
-      { configFile: './test_common/test.babel.js' },
-    ],
-  },
-  moduleNameMapper: {
-    ...jestModuleMapper,
-  },
-  setupFiles: ['<rootDir>/test_common/jest_cucumber_support/index.ts'],
-  globals: {
-    'ts-jest': {
-      tsconfig: '<rootDir>/test_common/tsconfig.test.json',
-    },
-  },
+  coverageProvider: 'v8',
```

Comment:
So...

If you use the default `babel` provider - it will not know how to provide coverage to the two projects as they have different tsconfig (one needs to support jsx + esnext, other is older js module).

If you set `v8` - it is able to hook into the compiled typescript itself. Coverage numbers seem to possibly be based upon the transpiled JS - but, I've found that if it says 100% line coverage it's telling the truth, but I don't know if the branches section is misleading.